stock_location_children/models/stock_location.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockLocation(models.Model):
+
+    _inherit = "stock.location"
+
+    children_ids = fields.Many2many(
+        "stock.location",
+        "stock_location_children_ids",
+        "parent_id",
+        "children_id",
+        compute="_compute_children_ids",
+        store=True,
+        help="All the children (multi-level) stock location of this location",
+    )
+
+    @api.depends("child_ids", "child_ids.child_ids")
+    def _compute_children_ids(self):
+        query = """SELECT sub.id, ARRAY_AGG(sl2.id) AS children
+            FROM stock_location sl2,
+            (
+            SELECT id, parent_path
+            FROM stock_location sl
+            ) sub
+            WHERE sl2.parent_path LIKE sub.parent_path || '%%'
+            AND sl2.id != sub.id
+            AND sub.id IN %s
+            GROUP BY sub.id;
+        """
+        self.env.cr.execute(query, (tuple(self.ids),))
```

Comment:
```suggestion
        self.flush()
        self.env.cr.execute(query, (tuple(self.ids),))
```

Reply:
It should not flush all the fields because it can flush too much (following relations and flushing other models, ...).
`flush("parent_id", "parent_path")`?

Reply:
Pushed a new commit